app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb

```diff
@@ -273,7 +273,7 @@ def volumes
 
   def get_formatted_tag(tag)
     formatted_key, formatted_value = tag[:name]&.split(":")
-    {:key => tag[:key] || formatted_key, :value => tag[:value] || formatted_value}
+    {:key => tag[:key] || formatted_key, :value => tag[:value] || formatted_value || ""}
```

Comment:
@Real-Omar-Afifi are you trying to prevent the value from being `nil` ?  I would have thought that was fine

Reply:
i thought so too, but we got errors and the refresh failed

Reply:
Interesting that feels like a bug, can you post the error that you're getting and I'll test with another provider

Reply:
@agrare 
evm.log
```
[----] E, [2020-10-28T15:52:57.888269 #28281:3fef64c2ce30] ERROR -- : MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh) EMS: [test-vpc], id: [5] Unable to perform refresh for the following targets:
[----] E, [2020-10-28T15:52:57.888357 #28281:3fef64c2ce30] ERROR -- : MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh)  --- ManageIQ::Providers::IbmCloud::VPC::CloudManager [test-vpc] id [5]
[----] I, [2020-10-28T15:52:57.924541 #28281:3fef64c2ce30]  INFO -- : MIQ(ManageIQ::Providers::IbmCloud::VPC::CloudManager::Refresher#refresh) Refreshing all targets...Complete
[----] E, [2020-10-28T15:52:57.924724 #28281:3fef64c2ce30] ERROR -- : MIQ(MiqQueue#deliver) Message id: [2119], Error: [undefined method `empty?' for nil:NilClass]
[----] E, [2020-10-28T15:52:57.925163 #28281:3fef64c2ce30] ERROR -- : [ManageIQ::Providers::BaseManager::Refresher::PartialRefreshError]: undefined method `empty?' for nil:NilClass  Method:[block (2 levels) in <class:LogProxy>]
[----] E, [2020-10-28T15:52:57.925252 #28281:3fef64c2ce30] ERROR -- : /Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:67:in `refresh'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:11:in `refresh'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/ems_refresh.rb:71:in `block in refresh'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/ems_refresh.rb:70:in `each'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/ems_refresh.rb:70:in `refresh'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue.rb:484:in `block in dispatch_method'
/Users/chadholliday/.rbenv/versions/2.5.8/lib/ruby/2.5.0/timeout.rb:93:in `block in timeout'
/Users/chadholliday/.rbenv/versions/2.5.8/lib/ruby/2.5.0/timeout.rb:33:in `block in catch'
/Users/chadholliday/.rbenv/versions/2.5.8/lib/ruby/2.5.0/timeout.rb:33:in `catch'
/Users/chadholliday/.rbenv/versions/2.5.8/lib/ruby/2.5.0/timeout.rb:33:in `catch'
/Users/chadholliday/.rbenv/versions/2.5.8/lib/ruby/2.5.0/timeout.rb:108:in `timeout'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue.rb:482:in `dispatch_method'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue.rb:459:in `block in deliver'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/user.rb:360:in `with_user_group'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue.rb:459:in `deliver'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue_worker_base/runner.rb:80:in `deliver_queue_message'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue_worker_base/runner.rb:104:in `deliver_message'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue_worker_base/runner.rb:122:in `block in do_work'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue_worker_base/runner.rb:118:in `loop'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_queue_worker_base/runner.rb:118:in `do_work'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_worker/runner.rb:264:in `block in do_work_loop'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_worker/runner.rb:261:in `loop'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_worker/runner.rb:261:in `do_work_loop'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_worker/runner.rb:113:in `run'
/Users/chadholliday/Development/manage-iq-dev/manageiq/app/models/miq_worker/runner.rb:95:in `start'
/Users/chadholliday/Development/manage-iq-dev/manageiq/lib/workers/bin/run_single_worker.rb:113:in `<main>'
```

and heres the stack trace of a vpc refresh when vcr is turned off to do a live api call.
```
     NoMethodError:
       undefined method `empty?' for nil:NilClass
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:119:in `map_name_type_value'
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:108:in `map_label'
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:71:in `block in map_labels'
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:71:in `each'
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:71:in `collect_concat'
     # ./spec/manageiq/app/models/provider_tag_mapping/mapper.rb:71:in `map_labels'
     # ./app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:160:in `map_labels'
     # ./app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:64:in `block in instances'
     # ./app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:44:in `each'
     # ./app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:44:in `instances'
     # ./app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb:21:in `parse'
     # ./spec/manageiq/app/models/manageiq/providers/inventory.rb:42:in `block in parse'
     # ./spec/manageiq/app/models/manageiq/providers/inventory.rb:39:in `each'
     # ./spec/manageiq/app/models/manageiq/providers/inventory.rb:39:in `parse'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:125:in `parse_targeted_inventory'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:92:in `block in refresh_targets_for_ems'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:91:in `refresh_targets_for_ems'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block (2 levels) in refresh'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:41:in `block in refresh'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:31:in `each'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:31:in `refresh'
     # ./spec/manageiq/app/models/manageiq/providers/base_manager/refresher.rb:11:in `refresh'
     # ./spec/manageiq/app/models/ems_refresh.rb:71:in `block in refresh'
     # ./spec/manageiq/app/models/ems_refresh.rb:70:in `each'
     # ./spec/manageiq/app/models/ems_refresh.rb:70:in `refresh'
     # ./spec/manageiq/app/models/ext_management_system.rb:610:in `refresh'
```